kuwala/core/canvas/src/components/Modals/TransformationCatalog/TransformationCatalogModal.js

```diff
@@ -0,0 +1,329 @@
+                </div>
+            )
+        } else {
+            return (
+                <div className={Classes.SelectedTransformationContainer}>
+                    <div className={Classes.TransformationOptionsContainer}>
+                        {renderTransformationOptions(catalogOptions)}
+                    </div>
+                    <div className={Classes.SelectedOptionDetailsContainer}>
+                        <div className={Classes.OptionDetailsContainer}>
+                            {renderSelectedOptionDetails()}
+                        </div>
+                    </div>
+                </div>
+            )
+        }
+    }
+
+    const renderTransformationOptions = (options) => {
+        return options.map(renderOptionItem)
+    }
+
+    const renderOptionItem = (optionItem, index) => {
+        return (
+            <Button
+                solid={false}
+                color={'kuwalaPurple'}
+                onClick={()=>{
+                    setSelectedCatalogOption(index)
+                }}
+            >
+                <div className={Classes.WideButtonContainer}>
+                    {getCatalogItemIcon(optionItem.icon)}
+                    <span className={'font-semibold'}>
+                        {optionItem.name}
+                    </span>
+                </div>
+            </Button>
+        )
+    }
+
+    const renderSelectedOptionDetails = () => {
+        if(selectedCatalogOption === null){
+            return (
+                <div className={Classes.OptionDetailsAlertText}>
+                    <p>To see the <span className={'text-kuwala-purple'}>details</span> of a transformation block,</p>
+                    <p>select one from the left</p>
+                </div>
+            )
+        } else {
+            const optionItem = catalogOptions[selectedCatalogOption];
+            return (
+                <div className={Classes.OptionDetailsContent}>
+                    <div className={'text-kuwala-purple space-x-4'}>
+                        {getCatalogItemIcon(optionItem.icon)}
+                        <span className={'font-semibold'}>
+                            {optionItem.name}
+                        </span>
+                    </div>
+                    <p className={'mt-2'}>
+                        {optionItem.description}
+                    </p>
+                    <div className={cn(Classes.RequiredColumnBase, 'mt-6')}>
+                        <span>
+                            Required column types:
+                        </span>
+                        {renderBadgeList(optionItem.required_column_types)}
+                    </div>
+                    <div className={cn(Classes.RequiredColumnBase, 'mt-2')}>
+                        <span>
+                            Optional column types:
+                        </span>
+                        {renderBadgeList(optionItem.optional_column_types)}
+                    </div>
```

Comment:
Required column types and optional column types should each only be rendered if there are items in the array.

Here for example, the field optional column types shouldn't be rendered:

![image](https://user-images.githubusercontent.com/28758040/165484128-1d47472e-51e9-419d-abd4-2ba9b427fd17.png)